packages/utils/src/lib/reports/formatting.unit.test.ts

```diff
@@ -126,100 +128,161 @@ describe('formatSourceLine', () => {
+  it('should format the file path as a link when outputDir is defined (VS Code)', () => {
+    vi.stubEnv('TERM_PROGRAM', 'vscode');
+    vi.stubEnv('GITHUB_ACTIONS', 'false');
+    const filePath = toUnixPath('/packages/utils/src/index.ts');
+    const outputDir = toUnixPath('/.code-pushup');
+
     expect(
-      linkToLocalSourceForIde(
-        { file: '/packages/utils/src/index.ts' },
-        { outputDir: '/.code-pushup' },
-      ).toString(),
+      linkToLocalSourceForIde({ file: filePath }, { outputDir }).toString(),
     ).toBe('[`/packages/utils/src/index.ts`](../packages/utils/src/index.ts)');
   });
 
-  it('should return link to specific line when startLine is provided (VS Code)', () => {
+  it('should return a link to a specific line when startLine is provided (VS Code)', () => {
     vi.stubEnv('TERM_PROGRAM', 'vscode');
+    vi.stubEnv('GITHUB_ACTIONS', 'false');
+    const filePath = toUnixPath('/packages/utils/src/index.ts');
+    const outputDir = toUnixPath('/.code-pushup');
+
     expect(
       linkToLocalSourceForIde(
-        {
-          file: '/packages/utils/src/index.ts',
-          position: { startLine: 2 },
-        },
-        { outputDir: '/.code-pushup' },
+        { file: filePath, position: { startLine: 2 } },
+        { outputDir },
       ).toString(),
     ).toBe(
       '[`/packages/utils/src/index.ts`](../packages/utils/src/index.ts#L2)',
     );
-    vi.unstubAllEnvs();
   });
 });
 
-describe('formatFilePosition', () => {
+describe('getGitHubLink', () => {
+  beforeEach(() => {
+    vi.stubEnv('TERM_PROGRAM', '');
+    vi.stubEnv('GITHUB_ACTIONS', 'true');
+    vi.stubEnv('GITHUB_SERVER_URL', 'https://github.com');
+    vi.stubEnv('GITHUB_REPOSITORY', 'user/repo');
+    vi.stubEnv('GITHUB_SHA', '1234567890abcdef');
+  });
+
   it.each([
-    ['../src/index.ts', { startLine: 2 }, '../src/index.ts#L2'],
-    ['../src/index.ts', { startLine: 2, endLine: 5 }, '../src/index.ts#L2-L5'],
-    ['../src/index.ts', { startLine: 2, startColumn: 1 }, '../src/index.ts#L2'],
+    [
+      { startLine: 2 },
+      'https://github.com/user/repo/blob/1234567890abcdef/.code-pushup/src/index.ts#L2',
+    ],
+    [
+      { startLine: 2, endLine: 5 },
+      'https://github.com/user/repo/blob/1234567890abcdef/.code-pushup/src/index.ts#L2-L5',
+    ],
+    [
+      { startLine: 2, startColumn: 1 },
+      'https://github.com/user/repo/blob/1234567890abcdef/.code-pushup/src/index.ts#L2',
+    ],
   ])(
-    'should transform file path "%s" by including position %o when running in GitHub',
-    (file, position, expected) => {
-      vi.stubEnv('GITHUB_ACTIONS', 'true');
-      vi.stubEnv('TERM_PROGRAM', '');
-      expect(formatFilePosition(file, position)).toBe(expected);
-      vi.unstubAllEnvs();
+    'should generate a GitHub repository link for the file with position %o',
+    (position, expected) => {
+      expect(
+        getGitHubLink(
+          toUnixPath('src/index.ts'),
```

Comment:
The URL is incorrect. The issue's file path is relative to the root git directory, not the report output directory, so it should be `https://github.com/user/repo/blob/1234567890abcdef/src/index.ts#L2`. The `outputDir` is actually irrelevant for the GitHub URL.

Reply:
Makes total sense! Don't know why I assumed `outputDir` was part of the GitHub link